C# script can't used dynamic

e.g. running the following C# http func:

```c#
using System;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Threading.Tasks;
using System.Diagnostics;
using Microsoft.Azure.WebJobs.Host;

public static HttpResponseMessage Run(HttpRequestMessage req, TraceWriter log)
{
    dynamic d = req;
    if (d.foo == null) {
        
    }

    var res = new HttpResponseMessage(HttpStatusCode.OK)
    {
        Content = new StringContent("Hello")
    };
    
    return res;
}
```

Fails with

    run.csx(12,9): error CS0656: Missing compiler required member 'Microsoft.CSharp.RuntimeBinder.CSharpArgumentInfo.Create'

Based on http://stackoverflow.com/questions/22864790/using-system-dynamic-in-roslyn, the fix is to reference `Microsoft.CSharp.dll`. I don't think we're carrying the assembly right now, so we may need to include it.

--- a/src/WebJobs.Script/Description/CSharp/FunctionMetadataResolver.cs
+++ b/src/WebJobs.Script/Description/CSharp/FunctionMetadataResolver.cs
@@ -36,6 +36,7 @@ namespace Microsoft.Azure.WebJobs.Script.Description
                 "System.Core",
                 "System.Xml",
                 "System.Net.Http",
+                "Microsoft.CSharp",
                 typeof(object).Assembly.Location,
                 typeof(IAsyncCollector<>).Assembly.Location, /*Microsoft.Azure.WebJobs*/
                 typeof(JobHost).Assembly.Location, /*Microsoft.Azure.WebJobs.Host*/

